File: TreasurersApp/Models/EmailTypeActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class EmailTypeActionResult : WebResult<EmailType>
    {
        public EmailTypeActionResult(bool success, List<string> statusMessages, EmailType emailType)
            :base(success, statusMessages, emailType)
        {
        }
    }
}

File: TreasurersApp/ClientApp/src/app/models/EmailTypeActionResult.ts
  import { EmailType } from './EmailType';
  import { EmailTypeService } from '../maintenance/email-type-maintenance/email-type.service';

  export class EmailTypeActionResult {
    success = false;
    statusMessages: string[] = [];
    emailType: EmailType = null;

    constructor(private emailTypeService: EmailTypeService) {
      this.success = false;
      this.emailType = this.emailTypeService.newEmailType();
      this.statusMessages = [];
    }
  }
